### DISCUSSION TITLE
ATtiny84 analogRead() for Differential Input channels

### DISCUSSION THREAD

**Discussion by rickg4tx** [SpenceKonde/ATTinyCore#645] [Support / Q & A]
What do I need to do setup Differential Input channels?  The documentation states to use the ADMUX values for the Differential Input channels.
I'm using PA0-PA5 for single ended ADC and want to use PA6 and PA7 for Differential Input channel.  This gives me four choices for analogRead():
 Differential Input channels

```
ADMUX        1x            20x
PA6+/PA7-  B011110 (30)  B011111 (31)
PA7+/PA6-  B111110 (62)  B111111 (63)
```

The following code always gives 6 for port values of 30,31,62,63 no mater what I connect to PA6 and PA7:
```
void loop() {
  if (a84Serial.available() > 0) {
    uint16_t adc = 0;
    uint8_t port = a84Serial.read();
    if (port < 8) {
      adc = analogRead(port);
    }
    else if (port == 30) {//1xDifferential Input A6+ A7-
      adc = analogRead(port); //B011110
    }
    else if (port == 31) {//20xDifferential Input A6+ A7-
      adc = analogRead(port); //B011111
    }
    else if (port == 62) {//1xDifferential Input A7+ A6-
      adc = analogRead(port); //B111110
    }
    else if (port == 63) {//20xDifferential Input A7+ A6-
      adc = analogRead(port); //B111111
    }
    a84Serial.write(adc);       //lsb
    a84Serial.write(adc >> 8);  //msb
  }
}
```

**Comment by SpenceKonde** [comment]
Wait for 2.0.0 to be released which has this functionality fully
integrated, it is my current project to get 2.0.0 out, because every issue
report that comes in I fixed in 5h3 past year already and it just isn't
available in a usable release

____________
Spence Konde
Azzy’S Electronics

New products! Check them out at tindie.com/stores/DrAzzy
GitHub: github.com/SpenceKonde
ATTinyCore: Arduino support for almost every ATTiny microcontroller
Contact: ***@***.***

**Reply by rickg4tx** [reply]
Thanks so much for such quick reply.  Please disregard my comments.  It seems you need to take a few readings before it settles down.  Much noisier than single ended ADC.  Your analogRead() seems to work as documented.
Thanks for your work!!!
